text/src/autogluon/text/automm/predictor.py

```diff
@@ -282,21 +286,39 @@ def fit(
             knowledge to a student predictor, i.e., the current predictor.
         seed
             The random seed to use for this training run.
+        hyperparameter_tune_kwargs
+                Hyperparameter tuning strategy and kwargs (for example, how many HPO trials to run).
+                If None, then hyperparameter tuning will not be performed.
+                    num_trials: int
+                        How many HPO trials to run. Either `num_trials` or `time_limit` to `fit` needs t o be specified.
+                    scheduler: Union[str, ray.tune.schedulers.TrialScheduler]
+                        If str is passed, AutoGluon will create the scheduler for you with some default parameters.
+                        If ray.tune.schedulers.TrialScheduler object is passed, you are responsible for initializing the object.
+                    scheduler_init_args: Optional[dict] = None
+                        If provided str to `scheduler`, you can optionally provide custom init_args to the scheduler
+                    searcher: Union[str, ray.tune.suggest.SearchAlgorithm, ray.tune.suggest.Searcher]
+                        If str is passed, AutoGluon will create the searcher for you with some default parameters.
+                        If ray.tune.schedulers.TrialScheduler object is passed, you are responsible for initializing the object.
+                        You don't need to worry about `metric` and `mode` of the searcher object. AutoGluon will figure it out by itself.
+                    scheduler_init_args: Optional[dict] = None
+                        If provided str to `searcher`, you can optionally provide custom init_args to the searcher
+                        You don't need to worry about `metric` and `mode`. AutoGluon will figure it out by itself.
 
         Returns
         -------
         An "AutoMMPredictor" object (itself).
         """
+        if hyperparameter_tune_kwargs is not None:
+            if teacher_predictor is not None:
+                assert isinstance(teacher_predictor, str), 'HPO with distillation only supports passing a path to the predictor'
+            if self._continuous_training:
```

Comment:
Consider adding a util function to control which hyper-parameters are tunable based on different modes.

Reply:
Added